src/ui/public/parse_query/lib/to_user.js

```diff
@@ -10,6 +10,7 @@ export default function toUser(text) {
   if (text == null) return '';
   if (_.isObject(text)) {
     if (text.query_string) return toUser(text.query_string.query);
+    if (text.match_all && !_.has(text.match_all, 'boost')) return '';
     return angular.toJson(text);
   }
   if (text === '*') {
```

Comment:
Good point.  I removed the lines.  Though the '*' is in not just saved searches, but saved dashboards and visualizations as well.  So pretty much all saved items will show '*' unless the user has added a specific query.